allennlp/models/archival.py

```diff
@@ -40,6 +37,10 @@ def archive_model(serialization_prefix: str,
         logger.error("weights file %s does not exist, unable to archive model", weights_file)
         return
 
+    config_file = os.path.join(serialization_prefix, "_model_params.json")
```

Comment:
The prepended underscore here is a holdover from when the "serialization prefix" was actually a prefix, not a directory.  It should be removed (and the other places we have underscores, like where we're saving stdout, stderr, and the logs).  Also, we should rename `serialization_prefix`...

Another issue is that if you're using a newer version of the code with an older serialized model, and the default parameters have changed (like recently happened with our tokenizer), the loading could break.  It'd be a bit more robust to parse the log file to get the _actual_ parameters that were used (including defaults).  But that's way too much work to worry about right now.

Reply:
I'll open issues for these things; feel free to fix them now if you want, or leave it for someone to do later.

Reply:
I'll get rid of the underscores and rename serialization prefix.

We had briefly discussed whether it even belonged in the config file or whether it should be a command line arg to train, any strong preference there ?

Reply:
I think it's probably cleaner to have it be an argument to `train`, not in the config file.  Trying to think of where else that affects...  `evaluate` certainly would have to be changed to match, and a lot of tests.  Seems like a big change.  If you want to do it in this PR, that's fine with me, but it might make sense to merge this first, then do that.

Reply:
actually, the current version of `evaluate` just uses the archived `model.tar.gz`, so it doesn't even need to know about serialization dir. so I don't think it's a huge change, but let me do it in another PR just to be safe